src/main/java/com/rk/rentacarapp/services/VehicleService.java

```diff
@@ -0,0 +1,26 @@
+package com.rk.rentacarapp.services;
+
+import com.rk.rentacarapp.model.Vehicle;
+import com.rk.rentacarapp.model.VehicleType;
+
+import java.util.List;
+
+public interface VehicleService {
+
+    List<Vehicle> findAllVehicles();
+
+    Vehicle findVehicleById(Long id);
+
+    Vehicle findVehicleByRegPlate(String regPlate);
+
+    List<Vehicle> findAllVehiclesByType(VehicleType vehicleType);
```

Comment:
AllVehicles are kind of redundant. This is a class VehicleService, so some class will call a method on a variable that will likely be called vehicleService.filterByType(vehicleType), I think it's clear from that and the return type being a List<Vehicle> that it will return all vehicles that fit the filter